packages/editor/src/lib/editor/shapes/ShapeUtil.ts

```diff
@@ -193,6 +200,43 @@ export abstract class ShapeUtil<T extends TLUnknownShape = TLUnknownShape> {
 		return this.handlesCache.get(shape.id) ?? EMPTY_ARRAY
 	}
 
+	/**
+	 * Get an array of outline segments for the shape. For most shapes,
+	 * this will be a single segment that includes the entire outline.
+	 * For shapes with handles, this might be segments of the outline
+	 * between each handle.
+	 *
+	 * @example
+	 *
+	 * ```ts
+	 * util.getOutlineSegments(myShape)
+	 * ```
+	 *
+	 * @param shape - The shape.
+	 * @public
+	 */
+	protected getOutlineSegments(shape: T): Vec2d[][] {
+		return [this.outline(shape)]
+	}
+
+	@computed
+	private get outlineSegmentsCache(): ComputedCache<Vec2d[][], TLShape> {
+		return this.editor.store.createComputedCache('outline-segments:' + this.type, (shape) => {
+			return this.getOutlineSegments!(shape as any)
+		})
+	}
+
+	/**
+	 * Get the cached outline segments (this should not be overridden!)
+	 *
+	 * @param shape - The shape.
+	 * @public
+	 */
+	outlineSegments(shape: T): Vec2d[][] {
+		if (!this.getOutlineSegments) return EMPTY_ARRAY
+		return this.outlineSegmentsCache.get(shape.id) ?? EMPTY_ARRAY
+	}
+
```

Comment:
Added in order to move logic about lines outside of the snap manager. I could imagine this could be useful for most shapes with handles.